### PR TITLE
Arrow Meta integration within Arrow Core

### PR DESCRIPTION
**Pull request by danimontoya**
Addresses part of #246 .
Migration of prelude module from meta into Arrow core (migrate everything but `Kinds`)

### CODE REVIEW

File: arrow-core-data/src/main/kotlin/arrow/core/Option.kt [hunk 1]
```diff
@@ -544,11 +544,14 @@ sealed class Option<out A> : OptionOf<A> {
...
-    "Some(${SA.run { it.show() }})"
-  })
+  fun show(SA: Show<A>): String = fold(
+    {
+      "None"
+    },
+    {
+      "Some(${SA.run { it.show() }})"
+    }
+  )
```
Comment (rachelcarmena) [inline arrow-core-data/src/main/kotlin/arrow/core/Option.kt:554]: Hi @danimontoya :raised_hands: I see several format changes. What if we move these format changes to a single PR (with just format changes) to detect what changes come from the integration in an easier way?

Comment (rachelcarmena) [reply arrow-core-data/src/main/kotlin/arrow/core/Option.kt:554]: If they come from the new version of KTLint so we could create a PR for that upgrade before.

Comment (danimontoya) [reply arrow-core-data/src/main/kotlin/arrow/core/Option.kt:554]: yeah, makes totally sense! I'll do that, thanks for the suggestion! 😄

File: arrow-annotations/build.gradle [hunk 2]
```diff
@@ -9,3 +9,17 @@ apply from: "$DOC_CREATION"
 dependencies {
     api "io.kindedj:kindedj:$KINDEDJ_VERSION"
 }
+
+compileKotlin {
+    kotlinOptions {
+        useIR = true
+        freeCompilerArgs += ["-Xallow-jvm-ir-dependencies"]
```
Comment (rachelcarmena) [inline arrow-annotations/build.gradle:16]: @danimontoya :raised_hands: according to the errors, it seems this arg should be provided for all the modules. To solve it, you can create a PR in `arrow` repository with a branch called `new-arrow-meta-integration` (the same name) and include it here: https://github.com/arrow-kt/arrow/blob/master/gradle/subproject.gradle You'll see `kotlinOptions` there. Then re-run these checks.

**Comment by raulraja** [comment]
On hold after 1.0

**Pull request closed without merging**